nginx-controller/nginx/ingress.go

```diff
@@ -1,15 +1,37 @@
 package nginx
 
 import (
+	"fmt"
+	"strings"
+
 	api_v1 "k8s.io/client-go/pkg/api/v1"
 	extensions "k8s.io/client-go/pkg/apis/extensions/v1beta1"
 )
 
 // IngressEx holds an Ingress along with Secrets and Endpoints of the services
 // that are referenced in this Ingress
 type IngressEx struct {
-	Ingress    *extensions.Ingress
-	TLSSecrets map[string]*api_v1.Secret
-	JWTKey     *api_v1.Secret
-	Endpoints  map[string][]string
+	Ingress       *extensions.Ingress
+	TLSSecrets    map[string]*api_v1.Secret
+	JWTKey        *api_v1.Secret
+	Endpoints     map[string][]string
+	EndpointsInfo map[string][]EndpointInfo
+}
+
+// TODO
+type EndpointInfo struct {
+	Address string
+	Info    map[string]interface{}
```

Comment:
It is better to have `Pod`, `Container`, `Labels` fields instead of  `map[string]interface{}`

Reply:
It will lead to ugly copy-paste code in nginx/configurator.go where variable maps are created, because I will not be able to iterate over variable names.

Reply:
I see your point. However, this approach is not desirable. In Configurator, you have the code like this:
```
		for _, variable := range []string{
			"namespace",
			"pod",
			"container",
			"pod_labels",
		} 
```
Basically, with such code, we're losing the benefits of static typing. This makes the code brittle and less maintainable. For example,  this is how EndpointInfos are generated in Controller :
```
Info: map[string]interface{}{
 +						"namespace":  pod.Namespace,
 +						"pod":        pod.Name,
 +						"container":  container.Name,
 +						"pod_labels": nginx.Labels(pod.Labels),
 +					},
```
If we make a change to Controller code, change `container` to `container_name` or remove `namespace` and compile the code,  but do not change the code Configurator, we will see no errors at the compilation time. If we had  those map  entries as fields of  EndpointInfo, the change that we made would lead to a compilation error.